frontend/src/hooks/useMeasure.ts

```diff
@@ -0,0 +1,54 @@
+// from react-use https://github.com/streamich/react-use/blob/master/src/useMeasure.ts
+// modified to debounce setRect call prevent rerenders during animation
+import { useLayoutEffect, useMemo, useState } from 'react'
+import { debounce } from 'lodash'
+
+export type UseMeasureRect = Pick<
+  DOMRectReadOnly,
+  'x' | 'y' | 'top' | 'left' | 'right' | 'bottom' | 'height' | 'width'
+>
+export type UseMeasureRef<E extends Element = Element> = (element: E) => void
+export type UseMeasureResult<E extends Element = Element> = [
+  UseMeasureRef<E>,
+  UseMeasureRect,
+]
+
+const defaultState: UseMeasureRect = {
+  x: 0,
+  y: 0,
+  width: 0,
+  height: 0,
+  top: 0,
+  left: 0,
+  bottom: 0,
+  right: 0,
+}
+
+export function useMeasure<E extends Element = Element>(): UseMeasureResult<E> {
+  const [element, ref] = useState<E | null>(null)
+  const [rect, setRect] = useState<UseMeasureRect>(defaultState)
+
+  const deboucedSetRect = useMemo(() => debounce(setRect, 100), [])
```

Comment:
nit: since we are adding a new hook, should we attempt to make it generalisable, i.e allow passing of debounce timeout (with a default)?

Reply:
definitely but since its use is currently limited to this one use case, i think its ok to leave it as it is. will be strightforward to add it in the future